packages/php/email-editor/src/Integrations/Core/class-initializer.php

```diff
@@ -97,49 +100,65 @@ public function allow_styles( ?array $allowed_styles ): array {
 	 */
 	public function update_block_settings( array $settings ): array {
 		if ( in_array( $settings['name'], self::ALLOWED_BLOCK_TYPES, true ) ) {
-			$settings['supports']['email'] = true;
+			$settings['supports']['email']     = true;
+			$settings['render_email_callback'] = array( $this, 'render_block' );
 		}
 
-		switch ( $settings['name'] ) {
+		return $settings;
+	}
+
+	/**
+	 * Returns the block content rendered by the block renderer.
+	 *
+	 * @param string            $block_content Block content.
+	 * @param array             $parsed_block Parsed block settings.
+	 * @param Rendering_Context $rendering_context Rendering context.
+	 * @return string
+	 */
+	public function render_block( string $block_content, array $parsed_block, Rendering_Context $rendering_context ): string {
+		if ( isset( $parsed_block['blockName'] ) ) {
+			$block_renderer = $this->get_block_renderer( $parsed_block['blockName'] );
+			return $block_renderer->render( $block_content, $parsed_block, $rendering_context );
+		}
+
+		return $block_content;
+	}
+
+	/**
+	 * Return an instance of Abstract_Block_Renderer by the block name.
+	 *
+	 * @param string $block_name Block name.
+	 * @return Abstract_Block_Renderer
+	 */
+	public function get_block_renderer( string $block_name ): Abstract_Block_Renderer {
```

Comment:
I think this method could be private. It is used only in the member method render_block.